src/Service/ServiceUtils.cpp

```diff
@@ -391,14 +396,18 @@ static ErrorMessageOr<fs::path> FindSymbolsFilePathElf(
 }
 
 ErrorMessageOr<fs::path> FindSymbolsFilePath(
-    const orbit_grpc_protos::GetDebugInfoFileRequest& request,
-    const std::vector<fs::path>& search_directories) noexcept {
+    const orbit_grpc_protos::GetDebugInfoFileRequest& request) noexcept {
   auto object_file_or_error = CreateObjectFile(request.module_path());
   if (object_file_or_error.has_error()) {
     return ErrorMessage(absl::StrFormat("Unable to create object file: %s",
                                         object_file_or_error.error().message()));
   }
 
+  std::vector<fs::path> search_directories{kHardcodedSearchDirectories};
+  search_directories.insert(search_directories.end(),
+                            request.additional_search_directories().begin(),
+                            request.additional_search_directories().end());
```

Comment:
This code is dangerous because it won't work when `additional_search_directories()` doesn't return a reference but a value. Better do safe it into a variable first:

```
const auto& additional_search_directories = request.additional_search_directories();
search_directories.insert(search_directories.end(), additional_search_directories.begin(), additional_search_directories.end());
```

Reply:
done